pyartifactory/objects/artifact.py

```diff
@@ -96,29 +97,32 @@ def deploy(
                 for file in files:
                     self.deploy(Path(f"{root}/{file}"), Path(f"{new_root}/{file}"))
         else:
-            if checksums:
-                # based on https://github.com/thenewnano/python-artifactory/commit/3c095f88f5e4153d1feffbf3769d038eb16b4016
+            if checksum_enabled:
                 artifact_check_sums = Checksums.generate(local_file)
-                headers = {
-                    "X-Checksum-Deploy": "true",
-                    "X-Checksum-Sha1": artifact_check_sums.sha1,
-                    "X-Checksum-Sha256": artifact_check_sums.sha256,
-                    "X-Checksum": artifact_check_sums.md5,
-                }
+                try:
+                    self._put(
+                        f"{artifact_path}",
+                        headers={
+                            "X-Checksum-Deploy": "true",
+                            "X-Checksum-Sha1": artifact_check_sums.sha1,
+                            "X-Checksum-Sha256": artifact_check_sums.sha256,
+                            "X-Checksum": artifact_check_sums.md5,
+                        },
+                    )
+                except requests.exceptions.HTTPError as error:
+                    if error.response.status_code == 404:
+                        message = (
+                            f"Artifact {artifact_path} does not exist in Artifactory,"
+                            f" content is expected to deploy by checksum"
+                        )
+                        logger.error(message)
+                        raise ArtifactNotFoundError(message)
+                    raise ArtifactoryError from error
             else:
-                headers = {}
-            try:
-                self._put(f"{artifact_path}", headers=headers)
-            except requests.exceptions.HTTPError as error:
-                if error.response.status_code == 404 and not checksums:
-                    logger.info("Artifact is not on server, content upload is expected in order to deploy the artifact")
-                    headers["X-Checksum-Deploy"] = "false"
-                    with local_file.open("rb") as stream:
-                        self._put(f"{artifact_path}", data=stream, headers=headers)
-                        logger.debug("Artifact %s successfully deployed", local_file)
-                else:
-                    raise
+                with local_file.open("rb") as stream:
+                    self._put(f"{artifact_path}", data=stream)
```

Comment:
self._put(route=artifact_folder.as_posix(), ...) instead because` artifact_path` is Union[Path, str]